### ISSUE TITLE
Unexpected behaviour from reset() from '@formkit/core' when inputs are using v-model...

### ISSUE THREAD

**Issue by garylake** [formkit/formkit#238]
**Reproduction**
FormKit Vue Playground link:
https://formkit.link/db9e1154a6b7457be0a8b6bb3bb38fe2

**Describe the issue**
I suspect this might be by design but I'm running into the following issue and I appreciate the outcome might be advice and not a fix.  

When a formkit element is binded to local state using v-model, and you then call reset() from @formkit/core, the input is taken back to a cached version of what was provided onMount of the form. This might seem desirable/ideal functionality, but in my real application (not the formkit playground above), my form is beind v-modeled to data that has come from a pinia store/firebase backend. 

Therefore, when I reset() the form, I want to honour the reactivity, not go back to the cached version.

For example in my real app, I am:
* Submitting the form
* Saving changes to firebase via my pinia store and crucially staying on the same page
* On success, calling @formkit/core reset() so that I can clear down state in the form such as data-submitted="true"
* BUT the form is being rolled back to the old value and only when I refresh does pinia push the new value in as the form re mounts.
* As my form is still v-model binded to my pinia state, I'd expect the form inputs to model that data, not roll back to a cached version. 

So I get what it's doing but I feel like @formkit/core reset() is maybe overreaching here and this only becomes apparent when you start using it in the context of stores and a backend. 

To recap: When I've called @formkit/core reset(), my pinia store and firebase back have 'new-value' but the FormKit form input now shows  'old-value'. When i hard refresh the page, FormKit is mounted with data from the pinia store and shows the 'new-value'. 

**Outcomes**
Again, I appreciate this might be what you want @formkit/core reset() to do, so if you can give me a way to reset the validation/submitted states only on the form, that would at least help in my use case. I really only need to programatically reset things like data-submitted but my pinia store needs to be honoured as I've just committed it.

**Comment by andrew-boyd** [comment]
@garylake Yes, `reset()` by default uses a cached copy of the initial form data. You can however supply your own. For example `reset('FormID', {})` would result in completely clearing the form values. In the case of Pinia you could do something like `reset('FormID', myFormDataObjectFromPinia)` to have reset apply the new values instead of the cached ones. Here's a playground that re-assigns the `ref` values from your playground.

https://formkit.link/f346a4b8c5ee05d92042996e1ebcb9e8

**Comment by garylake** [comment]
@andrew-boyd amazing, that absolutely did the trick and saved doing a nasty router.go() to hard reload the page 🙏🏻
